Use assert methods of `unittest.TestCase` class rather than bare `assert` statement in tests

If you're using stdlib's `unittest` package for writing tests, you shouldn't use "bare" `assert` statements. They are intended for different purposes - to provide [quick-fail validation](https://wiki.python.org/moin/UsingAssertionsEffectively) of function call preconditions. But even this usage is considered an anti-pattern in community.

The issue is that `assert` statement could be [stripped out by interpreter when you turn optimization on](https://docs.python.org/3/using/cmdline.html#cmdoption-o), so your _tests will never fail_. I'll cite docs here for conveniece:

> -O
Remove assert statements and any code conditional on the value of `__debug__`. Augment the filename for compiled (bytecode) files by adding `.opt-1` before the `.pyc` extension (see PEP 488). See also PYTHONOPTIMIZE.

Here is example.

We have simple test:

```python
from unittest import TestCase


class TestTruth(TestCase):
    def test_true_is_false_bare_assert(self):
        assert True is False

    def test_true_is_false_assert_method(self):
        self.assertEqual(True, False)
```

Let's run it!

```console
$ python3 -m unittest test_truth.py 
FF
======================================================================
FAIL: test_true_is_false_assert_method (test_truth.TestTruth)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/tosh/spikes/python-bare-assert/test_truth.py", line 9, in test_true_is_false_assert_method
    self.assertEqual(True, False)
AssertionError: True != False

======================================================================
FAIL: test_true_is_false_bare_assert (test_truth.TestTruth)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/tosh/spikes/python-bare-assert/test_truth.py", line 6, in test_true_is_false_bare_assert
    assert True is False
AssertionError

----------------------------------------------------------------------
Ran 2 tests in 0.000s

FAILED (failures=2)
```

As you would expect, they both fails.

But let's turn optimization now (note added `-O` switch)!

```console
$ python3 -O -m unittest test_truth.py 
F.
======================================================================
FAIL: test_true_is_false_assert_method (test_truth.TestTruth)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/tosh/spikes/python-bare-assert/test_truth.py", line 9, in test_true_is_false_assert_method
    self.assertEqual(True, False)
AssertionError: True != False

----------------------------------------------------------------------
Ran 2 tests in 0.000s

FAILED (failures=1)
```

As you'll see, now only one test fails - the one that uses assert method rather than `assert` statement.

So, if you decided to write tests using stdlib's `unittest` package, you must use assert methods. Another option will be to switch testing framework. For example, there is [`pytest`](https://docs.pytest.org/en/latest/) framework, which "overloads" `assert` statement.

Let's run our test with `pytest` (with optimization turned on):

```console
$ pip3 install pytest
$ python3 -O -m pytest -q test_truth.py
WARNING: assertions not in test modules or plugins will be ignored because assert statements are not executed by the underlying Python interpreter (are you using python -O?)
FF                                                                                         [100%]
============================================ FAILURES ============================================
___________________________ TestTruth.test_true_is_false_assert_method ___________________________

self = <test_truth.TestTruth testMethod=test_true_is_false_assert_method>

    def test_true_is_false_assert_method(self):
>       self.assertEqual(True, False)
E       AssertionError: True != False

test_truth.py:9: AssertionError
____________________________ TestTruth.test_true_is_false_bare_assert ____________________________

self = <test_truth.TestTruth testMethod=test_true_is_false_bare_assert>

    def test_true_is_false_bare_assert(self):
>       assert True is False
E       assert True is False

test_truth.py:6: AssertionError
==================================== short test summary info =====================================
FAILED test_truth.py::TestTruth::test_true_is_false_assert_method - AssertionError: True != False
FAILED test_truth.py::TestTruth::test_true_is_false_bare_assert - assert True is False
2 failed in 0.03s
```

Note that `pytest` even warns you about bare `assert` statements. It is also more relaxed in other handy ways by contrast to `unittest` and can run `unittest`-style tests without additional configuration.

--- a/test/metrics/spaces/TestSpaces.py
+++ b/test/metrics/spaces/TestSpaces.py
@@ -32,24 +32,24 @@ class TestSpaces(TestCase):
 
     def test_class_with_best_ident(self):
         lines = self.pattern.value(Path(self.dir_path, 'BestIdent.java'))
-        assert lines[-2] == 44 and lines[-1] == 113
+        self.assertTrue(lines[-2] == 44 and lines[-1] == 113)
 
     def test_class_without_left_spaces(self):
         lines = self.pattern.value(Path(self.dir_path, 'NoLeftSpaces.java'))
-        assert lines[-2] == 0 and lines[-1] == 55
+        self.assertTrue(lines[-2] == 0 and lines[-1] == 55)
 
     def test_class_without_right_spaces(self):
         lines = self.pattern.value(Path(self.dir_path, 'NoRightSpaces.java'))
-        assert lines[-2] == 57 and lines[-1] == 0
+        self.assertTrue(lines[-2] == 57 and lines[-1] == 0)
 
     def test_class_with_equal_spaces_number(self):
         lines = self.pattern.value(Path(self.dir_path, 'SameMean.java'))
-        assert lines[-2] == 4 and lines[-1] == 55
+        self.assertTrue(lines[-2] == 4 and lines[-1] == 55)
 
     def test_class_with_tabs_and_spaces(self):
         lines = self.pattern.value(Path(self.dir_path, 'SpacesAndTabs.java'))
-        assert lines[-2] == 8 and lines[-1] == 59
+        self.assertTrue(lines[-2] == 8 and lines[-1] == 59)
 
     def test_class_with_worst_ident(self):
         lines = self.pattern.value(Path(self.dir_path, 'WorstIdentation.java'))
-        assert lines[-2] == 20 and lines[-1] == 163
+        self.assertTrue(lines[-2] == 20 and lines[-1] == 163)

--- a/test/patterns/er_class/test_er_class.py
+++ b/test/patterns/er_class/test_er_class.py
@@ -11,48 +11,48 @@ class TestErClass(TestCase):
 
     def test_manager_in_middle(self):
         lines = self.er_class_finder.value(Path(self.dir_path, 'AnimatableSplitDimensionPathValue.java'))
-        assert lines == [12]
+        self.assertEqual(lines, [12])
 
     def test_controller_in_end(self):
         lines = self.er_class_finder.value(Path(self.dir_path, 'AnimatableTransform.java'))
-        assert lines == [12]
+        self.assertEqual(lines, [12])
 
     def test_one_normal_class(self):
         lines = self.er_class_finder.value(Path(self.dir_path, 'AuditEventModelProcessor.java'))
-        assert lines == []
+        self.assertEqual(lines, [])
 
     def test_two_classes_with_pattern(self):
         lines = self.er_class_finder.value(Path(self.dir_path, 'BaseKeyframeAnimation.java'))
-        assert lines == [18, 186]
+        self.assertEqual(lines, [18, 186])
 
     def test_class_parser(self):
         lines = self.er_class_finder.value(Path(self.dir_path, 'Configuration.java'))
-        assert lines == [3106]
+        self.assertEqual(lines, [3106])
 
     def test_another_normal_class(self):
         lines = self.er_class_finder.value(Path(self.dir_path, 'FillContent.java'))
-        assert lines == []
+        self.assertEqual(lines, [])
 
     def test_four_normal_classes(self):
         lines = self.er_class_finder.value(Path(self.dir_path, 'FJIterateTest.java'))
-        assert lines == []
+        self.assertEqual(lines, [])
 
     def test_two_distant_normal_classes(self):
         lines = self.er_class_finder.value(Path(self.dir_path, 'FJListProcedureRunner.java'))
-        assert lines == []
+        self.assertEqual(lines, [])
 
     def test_classes_in_comments(self):
         lines = self.er_class_finder.value(Path(self.dir_path, 'KeyProviderCryptoExtension.java'))
-        assert lines == []
+        self.assertEqual(lines, [])
 
     def test_classes_in_methods(self):
         lines = self.er_class_finder.value(Path(self.dir_path, 'OsSecureRandom.java'))
-        assert lines == []
+        self.assertEqual(lines, [])
 
     def test_normal_class(self):
         lines = self.er_class_finder.value(Path(self.dir_path, 'RectangleContent.java'))
-        assert lines == []
+        self.assertEqual(lines, [])
 
     def test_three_writers_one_reader(self):
         lines = self.er_class_finder.value(Path(self.dir_path, 'SequenceFile.java'))
-        assert lines == [837, 1478, 1538, 1684]
+        self.assertEqual(lines, [837, 1478, 1538, 1684])

--- a/test/patterns/force_type_casting_finder/test_force_type_casting_finder.py
+++ b/test/patterns/force_type_casting_finder/test_force_type_casting_finder.py
@@ -30,16 +30,16 @@ class ForceTypeCastingFinderTest(TestCase):
         pattern = force_type_casting_finder.ForceTypeCastingFinder()
         lines = pattern.value(
             os.path.dirname(os.path.realpath(__file__)) + '/1.java')
-        assert lines == [5]
+        self.assertEqual(lines, [5])
 
     def test_several_casts(self):
         pattern = force_type_casting_finder.ForceTypeCastingFinder()
         lines = pattern.value(
             os.path.dirname(os.path.realpath(__file__)) + '/2.java')
-        assert lines == [5, 11, 17]
+        self.assertEqual(lines, [5, 11, 17])
 
     def test_zero_lines(self):
         pattern = force_type_casting_finder.ForceTypeCastingFinder()
         lines = pattern.value(
             os.path.dirname(os.path.realpath(__file__)) + '/3.java')
-        assert lines == []
+        self.assertEqual(lines, [])

--- a/test/patterns/if_return_if_detection/test_if_return_if.py
+++ b/test/patterns/if_return_if_detection/test_if_return_if.py
@@ -32,24 +32,24 @@ class TestCountIfReturn(TestCase):
 
     def test_no_return_inside(self):
         lines = self.countifreturn.value(Path(self.dir_path, '2.java'))
-        assert lines == []
+        self.assertEqual(lines, [])
 
     def test_2nd_level_inside(self):
         lines = self.countifreturn.value(Path(self.dir_path, '1.java'))
-        assert lines == [6, 10]
+        self.assertEqual(lines, [6, 10])
 
     def test_nested_one_goodreturn(self):
         lines = self.countifreturn.value(Path(self.dir_path, '3.java'))
-        assert lines == []
+        self.assertEqual(lines, [])
 
     def test_nested_one_badreturn(self):
         lines = self.countifreturn.value(Path(self.dir_path, '4.java'))
-        assert lines == [6]
+        self.assertEqual(lines, [6])
 
     def test_withandwithout_returns(self):
         lines = self.countifreturn.value(Path(self.dir_path, '5.java'))
-        assert lines == [6, 14, 16, 18]
+        self.assertEqual(lines, [6, 14, 16, 18])
 
     def test_nomoreReturn_and_nested(self):
         lines = self.countifreturn.value(Path(self.dir_path, '6.java'))
-        assert lines == [10]
+        self.assertEqual(lines, [10])

--- a/test/patterns/implements_multi/test_implements_multi.py
+++ b/test/patterns/implements_multi/test_implements_multi.py
@@ -11,48 +11,48 @@ class TestImplementsMulti(TestCase):
 
     def test_one_class_with_types(self):
         lines = self.multi_finder.value(Path(self.dir_path, 'AnimatableSplitDimensionPathValue.java'))
-        assert lines == []
+        self.assertEqual(lines, [])
 
     def test_two_classes(self):
         lines = self.multi_finder.value(Path(self.dir_path, 'AnimatableTransform.java'))
-        assert lines == [12]
+        self.assertEqual(lines, [12])
 
     def test_implements_in_string(self):
         lines = self.multi_finder.value(Path(self.dir_path, 'AuditEventModelProcessor.java'))
-        assert lines == []
+        self.assertEqual(lines, [])
 
     def test_implements_with_parantheses(self):
         lines = self.multi_finder.value(Path(self.dir_path, 'BaseKeyframeAnimation.java'))
-        assert lines == []
+        self.assertEqual(lines, [])
 
     def test_implements_with_nested_parantheses(self):
         lines = self.multi_finder.value(Path(self.dir_path, 'Configuration.java'))
-        assert lines == [225]
+        self.assertEqual(lines, [225])
 
     def test_implements_multi_classes(self):
         lines = self.multi_finder.value(Path(self.dir_path, 'FillContent.java'))
-        assert lines == [29]
+        self.assertEqual(lines, [29])
 
     def test_implements_with_parantheses_multi(self):
         lines = self.multi_finder.value(Path(self.dir_path, 'FJIterateTest.java'))
-        assert lines == [601]
+        self.assertEqual(lines, [601])
 
     def test_implements_with_parantheses_before(self):
         lines = self.multi_finder.value(Path(self.dir_path, 'FJListProcedureRunner.java'))
-        assert lines == []
+        self.assertEqual(lines, [])
 
     def test_implements_in_comments(self):
         lines = self.multi_finder.value(Path(self.dir_path, 'KeyProviderCryptoExtension.java'))
-        assert lines == []
+        self.assertEqual(lines, [])
 
     def test_implements_multi(self):
         lines = self.multi_finder.value(Path(self.dir_path, 'OsSecureRandom.java'))
-        assert lines == [42]
+        self.assertEqual(lines, [42])
 
     def test_implements_three(self):
         lines = self.multi_finder.value(Path(self.dir_path, 'RectangleContent.java'))
-        assert lines == [22]
+        self.assertEqual(lines, [22])
 
     def test_implements_many(self):
         lines = self.multi_finder.value(Path(self.dir_path, 'SequenceFile.java'))
-        assert lines == [837]
+        self.assertEqual(lines, [837])

--- a/test/patterns/instanceof/test_instanceof.py
+++ b/test/patterns/instanceof/test_instanceof.py
@@ -32,36 +32,36 @@ class TestInstanceOf(TestCase):
 
     def test_empty(self):
         file = str(Path(self.cur_file_dir, 'Empty.java'))
-        assert len(self.testClass.value(file)) == 0
+        self.assertEqual(len(self.testClass.value(file)), 0)
 
     def test_instance_of(self):
         file = str(Path(self.cur_file_dir, 'InstanceOfSample.java'))
-        assert len(self.testClass.value(file)) == 1
+        self.assertEqual(len(self.testClass.value(file)), 1)
 
     def test_instance(self):
         file = str(Path(self.cur_file_dir, 'InstanceSample.java'))
-        assert len(self.testClass.value(file)) == 1
+        self.assertEqual(len(self.testClass.value(file)), 1)
 
     def test_instance_of_different_methods(self):
         file = str(Path(self.cur_file_dir, 'InstanceOfSampleDifferentMethods.java'))
-        assert len(self.testClass.value(file)) == 2
+        self.assertEqual(len(self.testClass.value(file)), 2)
 
     def test_instance_different_methods(self):
         file = str(Path(self.cur_file_dir, 'InstanceSampleDifferentMethods.java'))
-        assert len(self.testClass.value(file)) == 2
+        self.assertEqual(len(self.testClass.value(file)), 2)
 
     def test_instance_of_several(self):
         file = str(Path(self.cur_file_dir, 'InstanceOfSampleSeveral.java'))
-        assert len(self.testClass.value(file)) == 3
+        self.assertEqual(len(self.testClass.value(file)), 3)
 
     def test_instance_several(self):
         file = str(Path(self.cur_file_dir, 'InstanceSampleSeveral.java'))
-        assert len(self.testClass.value(file)) == 2
+        self.assertEqual(len(self.testClass.value(file)), 2)
 
     def test_instance_in_method_chaining(self):
         file = str(Path(self.cur_file_dir, 'InstanceSampleChain.java'))
-        assert len(self.testClass.value(file)) == 2
+        self.assertEqual(len(self.testClass.value(file)), 2)
 
     def test_both(self):
         file = str(Path(self.cur_file_dir, 'InstanceBoth.java'))
-        assert len(self.testClass.value(file)) == 2
+        self.assertEqual(len(self.testClass.value(file)), 2)

--- a/test/patterns/method_chaining/test_method_chaining.py
+++ b/test/patterns/method_chaining/test_method_chaining.py
@@ -32,47 +32,47 @@ class TestMethodChain(TestCase):
 
     def test_method_chain(self):
         lines = self.method_chain_finder.value(Path(self.dir_path, 'MethodChain.java'))
-        assert len(lines) == 1
+        self.assertEqual(len(lines), 1)
 
     def test_empty_method_chain(self):
         lines = self.method_chain_finder.value(Path(self.dir_path, 'EmptyMethodChain.java'))
-        assert len(lines) == 1
+        self.assertEqual(len(lines), 1)
 
     def test_chain_with_new_object(self):
         lines = self.method_chain_finder.value(Path(self.dir_path, 'MethodChainNewObjectMethods.java'))
-        assert len(lines) == 1
+        self.assertEqual(len(lines), 1)
 
     def test_method_chain_in_different_methods(self):
         lines = self.method_chain_finder.value(Path(self.dir_path, 'MethodChainInDifferentMethods.java'))
-        assert len(lines) == 2
+        self.assertEqual(len(lines), 2)
 
     def test_chain_in_nested_class(self):
         lines = self.method_chain_finder.value(Path(self.dir_path, 'MethodChainNestedClass.java'))
-        assert len(lines) == 1
+        self.assertEqual(len(lines), 1)
 
     def test_chain_in_anonymous_class(self):
         lines = self.method_chain_finder.value(Path(self.dir_path, 'MethodChainAnonymousClass.java'))
-        assert len(lines) == 1
+        self.assertEqual(len(lines), 1)
 
     def test_chain_in_anonymous_class_empty(self):
         lines = self.method_chain_finder.value(Path(self.dir_path, 'MethodChainAnonymousClassEmpty.java'))
-        assert len(lines) == 0
+        self.assertEqual(len(lines), 0)
 
     def test_several_chains(self):
         lines = self.method_chain_finder.value(Path(self.dir_path, 'MethodChainSeveral.java'))
-        assert len(lines) == 3
+        self.assertEqual(len(lines), 3)
 
     def test_chain_without_object_creating(self):
         lines = self.method_chain_finder.value(Path(self.dir_path, 'WithoutObjectCreating.java'))
-        assert len(lines) == 1
+        self.assertEqual(len(lines), 1)
 
     def test_nested_chain_with_this(self):
         lines = self.method_chain_finder.value(Path(self.dir_path, 'NestedChainWIthThis.java'))
-        assert len(lines) == 2
+        self.assertEqual(len(lines), 2)
 
     def test_nested_chain_with_simple_method_invocation(self):
         lines = self.method_chain_finder.value(Path(self.dir_path, 'NestedChainWithSimpleMethodInvocation.java'))
-        assert len(lines) == 2
+        self.assertEqual(len(lines), 2)
 
     def test_nested_chain_complicated_structure(self):
         """
@@ -80,16 +80,16 @@ class TestMethodChain(TestCase):
         with nested anonymous classes
         """
         lines = self.method_chain_finder.value(Path(self.dir_path, 'HolyMolyNestedChain.java'))
-        assert len(lines) == 3
+        self.assertEqual(len(lines), 3)
 
     def test_smallest_chain(self):
         lines = self.method_chain_finder.value(Path(self.dir_path, 'SmallestChain.java'))
-        assert len(lines) == 1
+        self.assertEqual(len(lines), 1)
 
     def test_fake_chain(self):
         lines = self.method_chain_finder.value(Path(self.dir_path, 'FakeChain.java'))
-        assert len(lines) == 0
+        self.assertEqual(len(lines), 0)
 
     def test_many_chains(self):
         lines = self.method_chain_finder.value(Path(self.dir_path, 'MachineLearningGetResultsIT.java'))
-        assert len(lines) > 300
+        self.assertGreater(len(lines), 300)

--- a/test/patterns/multiply_try/test_multiple_try.py
+++ b/test/patterns/multiply_try/test_multiple_try.py
@@ -32,28 +32,28 @@ class TestMultipleTry(TestCase):
 
     def test_simple(self):
         lines = self.method_chain_finder.value(Path(self.dir_path, 'Simple.java'))
-        assert lines == [2]
+        self.assertEqual(lines, [2])
 
     def test_large_file(self):
         lines = self.method_chain_finder.value(Path(self.dir_path, 'Large.java'))
-        assert lines == [706, 620]
+        self.assertEqual(lines, [706, 620])
 
     def test_try_inside_anonymous(self):
         lines = self.method_chain_finder.value(Path(self.dir_path, 'TryInsideAnomymous.java'))
-        assert lines == [5]
+        self.assertEqual(lines, [5])
 
     def test_try_inside_catch(self):
         lines = self.method_chain_finder.value(Path(self.dir_path, 'TryInsideCatch.java'))
-        assert lines == [2]
+        self.assertEqual(lines, [2])
 
     def test_try_inside_finally(self):
         lines = self.method_chain_finder.value(Path(self.dir_path, 'TryInsideFinaly.java'))
-        assert lines == [2]
+        self.assertEqual(lines, [2])
 
     def test_try_inside_try(self):
         lines = self.method_chain_finder.value(Path(self.dir_path, 'TryInsideTry.java'))
-        assert lines == [2]
+        self.assertEqual(lines, [2])
 
     def test_try_method_overloading(self):
         lines = self.method_chain_finder.value(Path(self.dir_path, 'TryMethodOverloading.java'))
-        assert lines == []
+        self.assertEqual(lines, [])

--- a/test/patterns/nested_blocks/test_nested_blocks.py
+++ b/test/patterns/nested_blocks/test_nested_blocks.py
@@ -33,30 +33,30 @@ class TestNestedBlocks(TestCase):
 
     def test_single_for_loop(self):
         file = str(Path(self.cur_file_dir, 'SingleFor.java'))
-        assert self.testClass.value(file) == [15, 19]
+        self.assertEqual(self.testClass.value(file), [15, 19])
 
     def test_nested_for_loops(self):
         file = str(Path(self.cur_file_dir, 'NestedFor.java'))
-        assert self.testClass.value(file) == [22]
+        self.assertEqual(self.testClass.value(file), [22])
 
     def test_for_loops_in_different_methods(self):
         file = str(Path(self.cur_file_dir, 'DifferentMethods.java'))
-        assert self.testClass.value(file) == [28]
+        self.assertEqual(self.testClass.value(file), [28])
 
     def test_for_loops_in_nested_class(self):
         file = str(Path(self.cur_file_dir, 'NestedForInNestedClasses.java'))
-        assert self.testClass.value(file) == [9]
+        self.assertEqual(self.testClass.value(file), [9])
 
     def test_for_loops_in_anonymous_class(self):
         file = str(Path(self.cur_file_dir, 'ForInAnonymousFile.java'))
-        assert self.testClass.value(file) == [19]
+        self.assertEqual(self.testClass.value(file), [19])
 
     def test_nested_no_nested_if(self):
         pattern = NestedBlocks(2, BlockType.IF)
         file = str(Path(self.cur_file_dir, 'NestedNoIF.java'))
-        assert pattern.value(file) == []
+        self.assertEqual(pattern.value(file), [])
 
     def test_nested_if(self):
         pattern = NestedBlocks(2, BlockType.IF)
         file = str(Path(self.cur_file_dir, 'NestedIF.java'))
-        assert pattern.value(file) == [21, 42]
+        self.assertEqual(pattern.value(file), [21, 42])

--- a/test/patterns/string_concat/test_concat_string.py
+++ b/test/patterns/string_concat/test_concat_string.py
@@ -32,19 +32,19 @@ class TestConcatString(TestCase):
 
     def test_concat_strings_in_print(self):
         lines = self.concat_finder.value(Path(self.dir_path, 'ConcatInPrint.java'))
-        self.assertEqual(lines,  [14])
+        self.assertEqual(lines, [14])
 
     def test_member_plus_string(self):
         lines = self.concat_finder.value(Path(self.dir_path, 'MemberPlusString.java'))
-        self.assertEqual(lines,  [111])
+        self.assertEqual(lines, [111])
 
     def test_multiple_concat(self):
         lines = self.concat_finder.value(Path(self.dir_path, 'MultipleConcat.java'))
-        self.assertEqual(lines,  [22])
+        self.assertEqual(lines, [22])
 
     def test_empty_case(self):
         lines = self.concat_finder.value(Path(self.dir_path, 'Nothing.java'))
-        self.assertEqual(lines,  [])
+        self.assertEqual(lines, [])
 
     def test_string_plus_member(self):
         lines = self.concat_finder.value(Path(self.dir_path, 'StringPlusMember.java'))
@@ -52,19 +52,19 @@ class TestConcatString(TestCase):
 
     def test_many_concats(self):
         lines = self.concat_finder.value(Path(self.dir_path, 'ManyConcats.java'))
-        self.assertEqual(lines,  [22, 23, 24, 25])
+        self.assertEqual(lines, [22, 23, 24, 25])
 
     def test_concat_in_different_methods(self):
         lines = self.concat_finder.value(Path(self.dir_path, 'DifferentMethods.java'))
-        self.assertEqual(lines,  [13, 27])
+        self.assertEqual(lines, [13, 27])
 
     def test_fake_operator_plus(self):
         lines = self.concat_finder.value(Path(self.dir_path, 'FakePlusOperator.java'))
-        self.assertEqual(lines,  [])
+        self.assertEqual(lines, [])
 
     def test_giant_file(self):
         lines = self.concat_finder.value(Path(self.dir_path, 'CliFlags.java'))
-        self.assertEqual(lines,  [
+        self.assertEqual(lines, [
             34, 35, 36, 37, 51, 53, 297, 298, 82, 83, 84, 85, 86, 87, 88, 197,
             90, 91, 92, 297, 211, 298, 96, 97, 98, 112, 113, 127, 128, 129, 130,
             131, 132, 146, 148, 149, 150, 151, 153, 155, 156, 157, 158, 159, 173,

--- a/test/patterns/string_concat/test_concat_string.py
+++ b/test/patterns/string_concat/test_concat_string.py
@@ -32,43 +32,46 @@ class TestConcatString(TestCase):
 
     def test_concat_strings_in_print(self):
         lines = self.concat_finder.value(Path(self.dir_path, 'ConcatInPrint.java'))
-        assert lines == [14]
+        self.assertEqual(lines,  [14])
 
     def test_member_plus_string(self):
         lines = self.concat_finder.value(Path(self.dir_path, 'MemberPlusString.java'))
-        assert lines == [111]
+        self.assertEqual(lines,  [111])
 
     def test_multiple_concat(self):
         lines = self.concat_finder.value(Path(self.dir_path, 'MultipleConcat.java'))
-        assert lines == [22]
+        self.assertEqual(lines,  [22])
 
     def test_empty_case(self):
         lines = self.concat_finder.value(Path(self.dir_path, 'Nothing.java'))
-        assert lines == []
+        self.assertEqual(lines,  [])
 
     def test_string_plus_member(self):
         lines = self.concat_finder.value(Path(self.dir_path, 'StringPlusMember.java'))
-        assert len(lines) == 1
+        self.assertEqual(len(lines), 1)
 
     def test_many_concats(self):
         lines = self.concat_finder.value(Path(self.dir_path, 'ManyConcats.java'))
-        assert lines == [22, 23, 24, 25]
+        self.assertEqual(lines,  [22, 23, 24, 25])
 
     def test_concat_in_different_methods(self):
         lines = self.concat_finder.value(Path(self.dir_path, 'DifferentMethods.java'))
-        assert lines == [13, 27]
+        self.assertEqual(lines,  [13, 27])
 
     def test_fake_operator_plus(self):
         lines = self.concat_finder.value(Path(self.dir_path, 'FakePlusOperator.java'))
-        assert lines == []
+        self.assertEqual(lines,  [])
 
     def test_giant_file(self):
         lines = self.concat_finder.value(Path(self.dir_path, 'CliFlags.java'))
-        assert lines == [
-            34, 35, 36, 37, 51, 53, 297, 298, 82, 83, 84, 85, 86, 87, 88, 197, 90, 91, 92, 297,
-            211, 298, 96, 97, 98, 112, 113, 127, 128, 129, 130, 131, 132, 146, 148, 149, 150,
-            151, 153, 155, 156, 157, 158, 159, 173, 174, 188, 189, 190, 191, 192, 193, 194, 197,
-            198, 199, 200, 204, 207, 208, 209, 211, 212, 214, 215, 216, 230, 231, 232, 233, 236, 239,
-            240, 241, 242, 243, 244, 245, 259, 260, 261, 262, 263, 264, 278, 279, 280,
-            299, 282, 283, 297, 298, 299, 300, 314, 315, 316, 317, 318, 319, 320, 321, 325, 326,
-            329, 331, 332, 333, 335, 336, 339, 340, 341, 343, 345, 347, 348, 349]
+        self.assertEqual(lines,  [
+            34, 35, 36, 37, 51, 53, 297, 298, 82, 83, 84, 85, 86, 87, 88, 197,
+            90, 91, 92, 297, 211, 298, 96, 97, 98, 112, 113, 127, 128, 129, 130,
+            131, 132, 146, 148, 149, 150, 151, 153, 155, 156, 157, 158, 159, 173,
+            174, 188, 189, 190, 191, 192, 193, 194, 197, 198, 199, 200, 204, 207,
+            208, 209, 211, 212, 214, 215, 216, 230, 231, 232, 233, 236, 239, 240,
+            241, 242, 243, 244, 245, 259, 260, 261, 262, 263, 264, 278, 279, 280,
+            299, 282, 283, 297, 298, 299, 300, 314, 315, 316, 317, 318, 319, 320,
+            321, 325, 326, 329, 331, 332, 333, 335, 336, 339, 340, 341, 343, 345,
+            347, 348, 349
+        ])

--- a/test/patterns/supermethod/test_supermethod.py
+++ b/test/patterns/supermethod/test_supermethod.py
@@ -32,30 +32,30 @@ class TestSuperMethod(TestCase):
 
     def test_empty(self):
         file = str(Path(self.cur_file_dir, 'Empty.java'))
-        assert len(self.testClass.value(file)) == 0
+        self.assertEqual(len(self.testClass.value(file)), 0)
 
     def test_instance_of(self):
         # It has 2 matches in anonymous class!
         file = str(Path(self.cur_file_dir, 'Anonymous.java'))
-        assert len(self.testClass.value(file)) == 2
+        self.assertEqual(len(self.testClass.value(file)), 2)
 
     def test_instance(self):
         file = str(Path(self.cur_file_dir, 'Simple.java'))
-        assert len(self.testClass.value(file)) == 1
+        self.assertEqual(len(self.testClass.value(file)), 1)
 
     def test_several(self):
         # It has 2 matches in anonymous class!
         file = str(Path(self.cur_file_dir, 'Several.java'))
-        assert len(self.testClass.value(file)) == 7
+        self.assertEqual(len(self.testClass.value(file)), 7)
 
     def test_nested_class(self):
         file = str(Path(self.cur_file_dir, 'NestedClass.java'))
-        assert len(self.testClass.value(file)) == 1
+        self.assertEqual(len(self.testClass.value(file)), 1)
 
     def test_constructor(self):
         file = str(Path(self.cur_file_dir, 'Constructor.java'))
-        assert len(self.testClass.value(file)) == 0
+        self.assertEqual(len(self.testClass.value(file)), 0)
 
     def test_complicated_constructor(self):
         file = str(Path(self.cur_file_dir, 'ComplicatedChainConstructor.java'))
-        assert len(self.testClass.value(file)) == 0
+        self.assertEqual(len(self.testClass.value(file)), 0)

--- a/test/patterns/this_finder/test_find_this.py
+++ b/test/patterns/this_finder/test_find_this.py
@@ -31,40 +31,40 @@ class TestFindThis(TestCase):
 
     def test_several(self):
         lines = self.pattern.value(self.cur_dir + '/several.java')
-        assert lines == [4, 10, 20]
+        self.assertEqual(lines, [4, 10, 20])
 
     def test_simple1(self):
         lines = self.pattern.value(self.cur_dir + '/double_this.java')
-        assert lines == [12]
+        self.assertEqual(lines, [12])
 
     def test_simple2(self):
         lines = self.pattern.value(self.cur_dir + '/init_block.java')
-        assert lines == []
+        self.assertEqual(lines, [])
 
     def test_simple22(self):
         lines = self.pattern.value(self.cur_dir + '/init_static_block.java')
-        assert lines == []
+        self.assertEqual(lines, [])
 
     def test_simple3(self):
         lines = self.pattern.value(self.cur_dir + '/autocloseable.java')
-        assert lines == [4, 14, 31]
+        self.assertEqual(lines, [4, 14, 31])
 
     def test_simple4(self):
         lines = self.pattern.value(self.cur_dir + '/one_line_this.java')
-        assert lines == [11]
+        self.assertEqual(lines, [11])
 
     def test_simple5(self):
         lines = self.pattern.value(self.cur_dir + '/one_line_usage.java')
-        assert lines == [12]
+        self.assertEqual(lines, [12])
 
     def test_simple6(self):
         lines = self.pattern.value(self.cur_dir + '/super.java')
-        assert lines == [12]
+        self.assertEqual(lines, [12])
 
     def test_simple7(self):
         lines = self.pattern.value(self.cur_dir + '/holy_moly_constructor.java')
-        assert lines == [47]
+        self.assertEqual(lines, [47])
 
     def test_simple9(self):
         lines = self.pattern.value(self.cur_dir + '/super_this.java')
-        assert lines == [15, 25, 51, 62, 76, 87, 101]
+        self.assertEqual(lines, [15, 25, 51, 62, 76, 87, 101])

--- a/test/patterns/var_decl_diff/test_var_decl_diff.py
+++ b/test/patterns/var_decl_diff/test_var_decl_diff.py
@@ -29,19 +29,19 @@ class VarMiddleTest(TestCase):
     def test_good_class(self):
         pattern = VarDeclarationDistance(lines_th=2)
         lines = pattern.value(os.path.dirname(os.path.realpath(__file__)) + '/1.java')
-        assert lines == []
+        self.assertEqual(lines, [])
 
     def test_bad_class(self):
         pattern = VarDeclarationDistance(lines_th=2)
         lines = pattern.value(os.path.dirname(os.path.realpath(__file__)) + '/2.java')
-        assert lines == [13]
+        self.assertEqual(lines, [13])
 
     def test_bad_class2(self):
         pattern = VarDeclarationDistance(lines_th=5)
         lines = pattern.value(os.path.dirname(os.path.realpath(__file__)) + '/3.java')
-        assert lines == [783, 969]
+        self.assertEqual(lines, [783, 969])
 
     def test_case_with_multiline_function_arguments(self):
         pattern = VarDeclarationDistance(lines_th=2)
         lines = pattern.value(os.path.dirname(os.path.realpath(__file__)) + '/4.java')
-        assert lines == [14, 18]
+        self.assertEqual(lines, [14, 18])

--- a/test/patterns/var_middle/test_var_middle.py
+++ b/test/patterns/var_middle/test_var_middle.py
@@ -29,29 +29,29 @@ class VarMiddleTest(TestCase):
     def test_good_class(self):
         pattern = VarMiddle()
         lines = pattern.value(os.path.dirname(os.path.realpath(__file__)) + '/1.java')
-        assert lines == []
+        self.assertEqual(lines, [])
 
     def test_bad_class(self):
         pattern = VarMiddle()
         lines = pattern.value(os.path.dirname(os.path.realpath(__file__)) + '/2.java')
-        assert lines == [9, 16]
+        self.assertEqual(lines, [9, 16])
 
     def test_case_with_multiline_method_declaration(self):
         pattern = VarMiddle()
         lines = pattern.value(os.path.dirname(os.path.realpath(__file__)) + '/3.java')
-        assert lines == []
+        self.assertEqual(lines, [])
 
     def test_case_with_empty_lines(self):
         pattern = VarMiddle()
         lines = pattern.value(os.path.dirname(os.path.realpath(__file__)) + '/4.java')
-        assert lines == []
+        self.assertEqual(lines, [])
 
     def test_case_autoclosable(self):
         pattern = VarMiddle()
         lines = pattern.value(os.path.dirname(os.path.realpath(__file__)) + '/5.java')
-        assert lines == [20, 21]
+        self.assertEqual(lines, [20, 21])
 
     def test_case_nested_class(self):
         pattern = VarMiddle()
         lines = pattern.value(os.path.dirname(os.path.realpath(__file__)) + '/6.java')
-        assert lines == [10, 22, 30, 33]
+        self.assertEqual(lines, [10, 22, 30, 33])

